vdirsyncer/storage/dav.py

```diff
@@ -317,6 +349,39 @@ def request(self, method, path, **kwargs):
             url = utils.compat.urlparse.urljoin(self.url, path)
         if self._session is None:
             self._session = requests.session()
+            if self._use_oauth2_google:
```

Comment:
I'd prefer if we have an entirely new storage type for Google access instead of patching that logic in here, but this is a major refactoring effort and I'd be willing to do it myself.